backend/sqlmapper/drivers/pg.go

```diff
@@ -102,34 +103,128 @@ func (s *pgStore) ColumnMetadata(q sqlmapper.Query) ([]database.Column, error) {
 	return q.ColumnMetadata(m.Columns)
 }
 
-func (s *pgStore) Create(tableName string, d sqlmapper.RowData) (sqlmapper.RowData, error) {
-	if err := verifyInput(d, tableName, s.modelMap); err != nil {
+func (s *pgStore) getRelationship(tableName string, relateTableName string) string {
+	relationships := s.modelMap[tableName].Relationship
+	var relationship string
+	for _, rel := range relationships {
+		if rel.Table == relateTableName {
+			relationship = rel.Type
+		}
+	}
+
+	return relationship
+}
+
+func (s *pgStore) getRelateColumn(tableName string, relateTableName string) (*database.Column, error) {
+	cs := s.modelMap[relateTableName].Columns
+	var c *database.Column
+	for i := 0; i < len(cs); i++ {
+		if cs[i].ForeignKey.Table == tableName {
+			c = &cs[i]
+			break
+		}
+	}
+
+	if c == nil {
+		return nil, errors.New("Can't find relate column")
+	}
+
+	return c, nil
+}
+
+func (s *pgStore) Create(tableName string, row sqlmapper.RowData) (sqlmapper.RowData, error) {
+	if err := verifyInput(row, tableName, s.modelMap); err != nil {
 		return nil, err
 	}
 
-	db := s.db.DB()
+	tx, err := s.db.DB().Begin()
+	if err != nil {
+		return nil, err
+	}
 
-	cols, data := d.ColumnsAndData()
+	cols, data := row.ColumnsAndData()
 
 	phs := strmangle.Placeholders(true, len(cols), 1, 1)
-
-	execQuery := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s) RETURNING id;",
+	sqlQuery := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s) RETURNING id;",
 		tableName,
 		strings.Join(cols, ","),
 		phs)
 
-	res := db.QueryRow(execQuery, data...)
+	stmt, err := tx.Prepare(sqlQuery)
+	if err != nil {
+		return nil, err
+	}
+	defer stmt.Close()
 
+	// update id if create success
 	var id int
-	err := res.Scan(&id)
+	err = stmt.QueryRow(data...).Scan(&id)
 	if err != nil {
+		err = tx.Rollback()
 		return nil, err
 	}
 
-	// update id if create success
-	d["id"] = sqlmapper.ColData{Data: id}
+	row["id"] = sqlmapper.ColData{Data: id}
 
-	return d, nil
+	// create relation data
+	relateRowData := row.RelateData()
+	for relateTableName, rows := range relateRowData {
+		relationship := s.getRelationship(tableName, relateTableName)
+
+		switch relationship {
+		case "has_many":
+			err = s.createWithHasMany(tx, tableName, id, relateTableName, rows)
+			if err != nil {
+				err = tx.Rollback()
+				return nil, err
+			}
+		default:
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	return row, nil
+}
+
+func (s *pgStore) createWithHasMany(tx *sql.Tx, tableName string, id int, relateTableName string, rows []sqlmapper.RowData) error {
```

Comment:
the `id` should have a more descriptive name

Reply:
because in current context is have many we can naming base on that relation, not need to have general name like `tableName`, `relateTableName`,...